open-morello-serial-ports.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/python3
+import argparse
+import logging
+import os
+import shlex
+import shutil
+import sys
+
+from enum import Enum
+
+from serial.tools.list_ports import comports  # noqa: E402
+from serial.tools.list_ports_common import ListPortInfo  # noqa: E402
+
+
+def find_morello_board_ttys(pretend: bool) -> "list[ListPortInfo]":
+    # find the serial port:
+    # Run `ioreg -p IOUSB -l -w 0` on macOS to find the right VID/PID
+    expected_vendor_id = 0x0403
+    expected_product_id = 0x6011
+    ttys = []
+    for portinfo in comports(include_links=True):
+        logging.debug("Checking %s", portinfo)
+        assert isinstance(portinfo, ListPortInfo)
+        if portinfo.pid == expected_product_id and portinfo.vid == expected_vendor_id:
+            ttys.append(portinfo)
```

Comment:
Does the `list(sorted(ttys, key=lambda x: (x.location, x.device)))` line below not work for you? I admit I only tested this on macOS where that gives me the expected order.

Reply:
I missed my opportunity to test with multiple boards attached as they are now in distinct locations. Maybe we don't need to support that for the initial version of this script?

Reply:
I've seen at least Linux do enough parallel enumeration that we'll get two boards sets of UARTS as X0-X3 Y0-Y3 X4-X7 Y4-Y7 rather than X0-X7 Y0-Y7.  But I agree that it's probably not a required feature; so maybe just a note about the `.location` should it become necessary.

Reply:
Oh, sorry, I missed what you were meaning about the `sorted` line below.  Yes, that probably works fine.